src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs

```diff
@@ -136,19 +140,25 @@ void MaybeMoveToNextRange(DateTime ts)
                 if (rangeSpec.WithinRange(ts))
                     return;
 
-                if (aggStates[0].Any)
+                if (interpolationType != InterpolationType.None)
                 {
-                    array.Add(AddTimeSeriesResult(timeSeriesFunction, aggStates, rangeSpec.Start, rangeSpec.End));
+                    HandleGapsIfNeeded(ts);
                 }
 
-                rangeSpec.MoveToNextRange(ts);
+                if (aggStates[0].Any)
+                {
```

Comment:
Why not to fill the gaps before `AddTimeSeriesResult`? we keep the track of previous stats and here we can check and fill if needed.

Reply:
what do you mean? we fill the gaps before `AddTimeSeriesResult`